pandas/_libs/tslib.pyx

```diff
@@ -664,11 +664,11 @@ cpdef array_to_datetime(
 
                         # Still raise OutOfBoundsDatetime,
                         # as error message is informative.
-                        raise
+                        raise OutOfBoundsDatetime(f"Cannot convert \"{val}\" at position {i} to datetime") from ex
```

Comment:
I think `ex`'s message should just be modified with the position information instead of raising a separate `OutOfBoundsDatetime`

Reply:
thanks @mroeschke for the clarification. ( apologies, had misunderstood the earlier review comment )
Will work on this change. 
one thing to note is that the functions (`check_dts_bounds`) which generate these exceptions don't have the position related info & hence some changes in function prototype will be needed for passing the position information to these functions.